sympy/functions/special/gamma_functions.py

```diff
@@ -657,95 +663,23 @@ class polygamma(Function):
-
-    def fdiff(self, argindex=2):
-        if argindex == 2:
-            n, z = self.args[:2]
-            return polygamma(n + 1, z)
-        else:
-            raise ArgumentIndexError(self, argindex)
-
-    def _eval_is_real(self):
-        if self.args[0].is_positive and self.args[1].is_positive:
-            return True
-
-    def _eval_is_complex(self):
-        z = self.args[1]
-        is_negative_integer = fuzzy_and([z.is_negative, z.is_integer])
-        return fuzzy_and([z.is_complex, fuzzy_not(is_negative_integer)])
-
-    def _eval_is_positive(self):
-        if self.args[0].is_positive and self.args[1].is_positive:
-            return self.args[0].is_odd
-
-    def _eval_is_negative(self):
-        if self.args[0].is_positive and self.args[1].is_positive:
-            return self.args[0].is_even
-
-    def _eval_aseries(self, n, args0, x, logx):
-        from sympy.series.order import Order
-        if args0[1] != oo or not \
-                (self.args[0].is_Integer and self.args[0].is_nonnegative):
-            return super()._eval_aseries(n, args0, x, logx)
-        z = self.args[1]
-        N = self.args[0]
-
-        if N == 0:
-            # digamma function series
-            # Abramowitz & Stegun, p. 259, 6.3.18
-            r = log(z) - 1/(2*z)
-            o = None
-            if n < 2:
-                o = Order(1/z, x)
-            else:
-                m = ceiling((n + 1)//2)
-                l = [bernoulli(2*k) / (2*k*z**(2*k)) for k in range(1, m)]
-                r -= Add(*l)
-                o = Order(1/z**n, x)
-            return r._eval_nseries(x, n, logx) + o
-        else:
-            # proper polygamma function
-            # Abramowitz & Stegun, p. 260, 6.4.10
-            # We return terms to order higher than O(x**n) on purpose
-            # -- otherwise we would not be able to return any terms for
-            #    quite a long time!
-            fac = gamma(N)
-            e0 = fac + N*fac/(2*z)
-            m = ceiling((n + 1)//2)
-            for k in range(1, m):
-                fac = fac*(2*k + N - 1)*(2*k + N - 2) / ((2*k)*(2*k - 1))
-                e0 += bernoulli(2*k)*fac/z**(2*k)
-            o = Order(1/z**(2*m), x)
-            if n == 0:
-                o = Order(1/z, x)
-            elif n == 1:
-                o = Order(1/z**2, x)
-            r = e0._eval_nseries(z, n, logx) + o
-            return (-1 * (-1/z)**N * r)._eval_nseries(x, n, logx)
-
     @classmethod
     def eval(cls, n, z):
         if n.is_integer:
             if n.is_nonnegative:
                 nz = unpolarify(z)
                 if z != nz:
                     return polygamma(n, nz)
-
             if n.is_positive:
                 if z is S.Half:
                     return S.NegativeOne**(n + 1)*factorial(n)*(2**(n + 1) - 1)*zeta(n + 1)
-
             if n is S.NegativeOne:
-                return loggamma(z)
+                return loggamma(z) - log(2*pi) / 2
```

Comment:
Is this a change in convention or does it just fix a wrong result?

Reply:
I tracked the current code down to #543 but that's such a massive PR that it's hard to understand exactly what's going on.

Reply:
It is indeed a change in convention. As noted in the first commit message, when Mathematica runs `PolyGamma[-1, x]` it uses a constant of integration of $0$ in $\psi(-1,x)=\int\psi(0,x)+K$, but that does not lead to a "balanced" function as defined by Espinosa & Moll.